drift/lib/src/sentry_query_interceptor.dart

```diff
@@ -0,0 +1,164 @@
+// ignore_for_file: invalid_use_of_internal_member
+
+import 'dart:async';
+
+import 'package:drift/drift.dart';
+import 'package:meta/meta.dart';
+import 'package:sentry/sentry.dart';
+
+import 'constants.dart' as drift_constants;
+import 'sentry_span_helper.dart';
+import 'version.dart';
+
+/// A Sentry query interceptor that wraps database operations in performance monitoring spans.
+///
+/// This interceptor tracks all database operations executed through a Drift database connection,
+/// including transactions, batches, and individual CRUD operations. Each operation is captured
+/// as a Sentry span with relevant context.
+class SentryQueryInterceptor extends QueryInterceptor {
+  final String _dbName;
+  late final SentrySpanHelper _spanHelper;
+  bool _isDbOpen = false;
+
+  SentryQueryInterceptor({required String databaseName, @internal Hub? hub})
+      : _dbName = databaseName {
+    _spanHelper = SentrySpanHelper(
+      SentryTraceOrigins.autoDbDriftQueryInterceptor,
+      hub: hub,
+    );
+    final options = hub?.options;
+    options?.sdk.addIntegration(drift_constants.integrationName);
+    options?.sdk.addPackage(packageName, sdkVersion);
+  }
+
+  /// Wraps database operations in Sentry spans.
+  ///
+  /// This handles most CRUD operations but excludes transaction lifecycle methods
+  /// (begin/commit/rollback), which require maintaining an ongoing transaction span
+  /// across multiple operations. Those are handled separately via [SentrySpanHelper].
+  Future<T> _instrumentOperation<T>(
+    String description,
+    FutureOr<T> Function() execute, {
+    String? operation,
+  }) async =>
+      _spanHelper.asyncWrapInSpan<T>(
+        description,
+        () async => execute(),
+        dbName: _dbName,
+        operation: operation,
+      );
+
+  @override
+  Future<bool> ensureOpen(QueryExecutor executor, QueryExecutorUser user) {
+    if (!_isDbOpen) {
+      _isDbOpen = true;
```

Comment:
Technically we set `_isDbOpen = true` before the db is open, as this is only the case when `super.ensureOpen(executor, user)` completes. Does this bother us?

Reply:
Before we would await fore setting the flag.

<img width="551" alt="Bildschirmfoto 2025-02-14 um 10 18 35" src="https://github.com/user-attachments/assets/6c2640af-f724-4086-9eb9-04b6d52fe152" />

Reply:
yeah I think the previous method is more consistent